src/arch/riscv/isa/formats/standard.isa

```diff
@@ -394,7 +433,7 @@ def template CSRExecute {{
             }
             DPRINTF(RiscvMisc, "Writing %#x to CSR %s.\n",
                     data, csrName);
-            xc->setMiscReg(midx, data);
+            isa->writeCSR(csr, data);
```

Comment:
It may affect the statistics of MiscReg if the `writeCSR` doesn't set miscReg value via `xc->setMiscReg`

https://github.com/gem5/gem5/blob/d2c8754ab3b72e52a07d62662de0120b941f1ae5/src/cpu/simple/exec_context.hh#L259-L263

Reply:
I think all paths of writeCSR result in setMiscReg being called